### PR TITLE
feat: add a layout service for converting dynamic flow steps without layouts to DynamicLayout format

### PR DESCRIPTION
**Pull request by StevePole**
## 🖼 Context

Layout is not a required property of dynamic flow steps so we need to handle layout of two different approaches.  Rather than do this, we will simply map steps without layouts to layout format using a service.  From then on, we can treat all dynamic flow steps as layouts and we need only one approach for the layout of our template.

## 🚀 Changes

We add a service that maps three step types (form, decision & final) to a dynamic layout specification

## 🤔 Considerations

The DynamicFlow component that will use the service is yet to be written.

We do not mimic the logic from V2 that narrows/widens the form based on the width of the fields inside the form, as in V3, it is not straightforward to interrogate the fields. 

We do not convert "upload steps" into side by side layouts, as in V2.  The intent is to add a layout to the step to achieve these more complex layouts.

## ✅ Checklist

- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/master/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+        data: {
+          exitValue: 'value',
+        },
+      };
+
+      const finalStep = {
+        type: 'final',
+        key: 'thing-final',
+        title: 'We create the thing!',
+        description: 'You now do stuff with the thing',
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/flow/layoutService/LayoutService.spec.js:81]: 💪

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.js [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    layout.push(convertStepDecisionToDynamicDecision(step.options));
+  }
+
+  return layout;
+}
+
+function dynamicBox(components, size) {
+  return {
+    type: 'box',
+    width: size || 'sm',
```
Comment (helendurranttw) [inline packages/dynamic-flows/src/flow/layoutService/LayoutService.js:74]: Should this default to sm? Not md?

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+function convertFormStepToDynamicLayout(step) {
+  let layout = convertCommonComponents(step);
+
+  if (step.reviewFields) {
+    layout.push(convertStepReviewToDynamicReview(step.reviewFields));
+  }
+
+  if (step.schemas) {
+    layout = layout.concat(getSchemaLayout(step));
```
[the remark this answers is not in the document]
Comment (StevePole) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.js:39]: Love it, that's what I was looking for and couldn't think of :-)

File: packages/dynamic-flows/src/flow/layoutService/LayoutService.js (hunk 2 above)
Comment (StevePole) [reply packages/dynamic-flows/src/flow/layoutService/LayoutService.js:74]: Yeah, that's fair.  I only went with that as my first use case needed 'sm', I subsequently extended it with an override. 'md' is a better default.

**Pull request merged**